src/js/actions/ui.js

```diff
@@ -440,13 +443,16 @@ define(function (require, exports) {
                 .bind(this)
                 .then(function () {
                     return this.transfer(centerBounds, document.bounds, false);
+                })
+                .then(function () {
+                    return this.transfer(tools.resetBorderPolicies);
```

Comment:
It seems like this does _not_ need to wait for `centerBounds` above? Could those be `Promise.join`'d in the same handler?